wallet/src/repository/wallets/mod.rs

```diff
@@ -106,11 +106,9 @@ impl<T: Database> Wallets<T> {
         let wallet_id = id.to_string();
         let lock = self.wallets_mutex.lock()?;
         let mut ids: Vec<String> = self.db.get_or_default(&keys::wallet_ids())?;
-        if !ids.contains(&wallet_id) {
```

Comment:
Maybe this was used to avoid having duplicate keys in `ids`?

Check that ids does not grow every time we re-create the same wallet:

```
ids: []
// Create wallet 111
ids: [111]
// Re-create wallet 111
ids: [111, 111]
// Re-create wallet 111
ids: [111, 111, 111]
```

Reply:
Ups...agree, the problem was the write batch being inside that condition!